hack/import-flux2-assets.sh

```diff
@@ -1,12 +1,19 @@
-#!/bin/sh -x
+#!/usr/bin/env bash
+
+set -euxo pipefail
 
 COMPONENTS_DIR="content/en/flux/components"
 FLUX_DIR="content/en/flux/cmd"
 
-if [ -z "${GITHUB_USER}" ]; then
+if [ -z "${GITHUB_USER:-}" ]; then
     GITHUB_USER=fluxcdbot
 fi
 
+if [ -z "${GITHUB_TOKEN:-}" ] ; then
+  echo "Please set GITHUB_TOKEN"
```

Comment:
Actually, I don't see why this requirement should be changed. The script works fine w/o the token set.

Reply:
In the flux GitHub action is the history for this, without a token you will get rate limited looking at the latest released version of a repo, and possibly some other things. This becomes especially important when the script runs in a GHA hosted runner, where it can be much more likely to get rate limited depending on what your neighbors are doing.

I'm not entirely sure why it's needed here, but in scripts like these in recent history that is why we've been adding these, (seek and destroy flaky tests.)

Reply:
Yes, I got rate-limited myself during development on the script because I haven't had GITHUB_TOKEN set. But it hasn't been mandatory so far and I'm not sure we should make it so because users running the script once or twice will not get rate-limited so quickly (well, there's NAT and stuff and I'm not sure how smart GitHub is with rate-limiting individual clients, but still, it worked so far).

Reply:
If you don't pass a token, the rate limit is based on your IP, (which is why it's so bad for runners, in general, because you will *always* reuse an IP address that's probably getting loads of activity.)

Reply:
Yes, but our GHA workflows all use a token, so enforcing it would rather be just another obstacle for local development, wouldn't it?